rotkehlchen/chain/evm/decoding/interfaces.py

```diff
@@ -171,8 +171,64 @@ def _decode_vote_cast(self, context: DecoderContext) -> DecodingOutput:
 
 class ReloadableDecoderMixin(metaclass=ABCMeta):
 
+    def __init__(
+            self,
+            cache_type_to_check_for_freshness: CacheType,
+            query_data_method: Callable,
+            save_data_to_cache_method: Callable,
+            read_pools_and_gauges_from_cache_method: Callable,
```

Comment:
not sure if you can add more information to the types in the return part of the callable.

```suggestion
            read_pools_and_gauges_from_cache_method: Callable[[], [list[str], list[str]]],
```

Reply:
I'm not sure about the value of data types for the other callables here. mypy seems to not handle them very well yet. it requires type:ignone wehn passing the methods as arguments. 
![image](https://github.com/rotki/rotki/assets/12560596/66588bb8-fc82-42ba-b7f8-7b2cedbf52d9)

Reply:
This seems wrong. mypy handles callable arguments quite well. But I guess you guys figured it out later? Please check mypy errors very well. type: ignore is only for mypy errors and bugs or things we can't fix yet.